### PR TITLE
Add docstrings.

### PR DESCRIPTION
**Pull request by knighton**

### CODE REVIEW

File: streaming/base/format/json/reader.py [hunk 1]
```diff
@@ -54,6 +54,16 @@ def __init__(
 
     @classmethod
     def from_json(cls, dirname: str, split: Optional[str], obj: Dict[str, Any]) -> Self:
+        """Initialize from JSON object.
+
+        Args:
+            dirname (str): Local directory containing shards.
+            split (Optional[str]): Which dataset split to use, if any.
```
Comment (karan6181) [inline streaming/base/format/json/reader.py:61]: Is there any standard or norm do have to follow between `(Optional[str])` and `(str, optional)` ? Do you know of any standardization?

File: streaming/base/format/json/writer.py [hunk 2]
```diff
@@ -11,6 +11,19 @@
 
 
 class JSONWriter(SplitWriter):
+    """Writes a streaming JSON dataset.
+
+    Args:
+        dirname (str): Local dataset directory.
```
Comment (karan6181) [inline streaming/base/format/json/writer.py:17]: Did you compile and ran the doc locally to see if it renders properly?

File: streaming/vision/cifar10.py [hunk 3]
```diff
@@ -5,4 +5,29 @@
 
 
 class CIFAR10(ImageClassDataset):
-    pass
+    """Streaming ImageNet.
```
Comment (karan6181) [inline streaming/vision/cifar10.py:8]: nit: CIFAR10

**Review by karan6181 — COMMENTED** [review]
LGTM. Thanks!

How did you format the doc? Did you ran `make style` and `make longlines` ? And can it be done automatically as part of `pre-commit run` ?

**Comment by knighton** [comment]
There are currently three subtly incompatible ways: (a) make targets, (b) github actions, (c) pre-commit hooks. Working on uniting them. Did the docstring formatting manually

File: streaming/base/format/json/reader.py (hunk 1 above)
Comment (knighton) [reply streaming/base/format/json/reader.py]: Fixed, everywhere

File: streaming/vision/cifar10.py (hunk 3 above)
Comment (knighton) [reply streaming/vision/cifar10.py]: Done

File: streaming/base/format/json/writer.py (hunk 2 above)
Comment (knighton) [reply streaming/base/format/json/writer.py:17]: https://mosaicml-composer--14.com.readthedocs.build/projects/streaming/en/14/

**Review by karan6181 — APPROVED** [review]
LGTM. Thanks!

**Pull request merged**